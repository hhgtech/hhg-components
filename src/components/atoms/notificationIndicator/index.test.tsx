import React from 'react'
import { render, cleanup, screen } from '@testing-library/react'
import 'jest-styled-components'

import { NotificationIndicator } from '.'

import MockTheme from './../../../mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const notificationIndicator = render(
    <MockTheme>
      <NotificationIndicator
        hasIcon={true}
        type="small"
        bellIcon={<img src="https://source.unsplash.com/random/16x16" />}
      />
    </MockTheme>,
  )

  const self = notificationIndicator.container.firstChild

  expect(self).toMatchSnapshot()

  expect(self).toHaveAttribute('data-has-icon', 'true')
  expect(self).toHaveAttribute('data-notification-type', 'small')
})
