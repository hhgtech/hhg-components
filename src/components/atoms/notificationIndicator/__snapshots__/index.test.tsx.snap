// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  position: relative;
}

.c0 span {
  background: #fe921d;
  border-radius: 50%;
  color: white;
  font-weight: 700;
  text-align: center;
}

.c0[data-notification-type='small'] span {
  display: block;
  width: 8px;
  height: 8px;
  font-size: 0px;
}

.c0[data-notification-type='medium'] span {
  display: block;
  width: 17px;
  height: 16px;
  font-size: 11px;
  line-height: 15px;
}

.c0[data-notification-type='large'] span {
  display: block;
  width: 24px;
  height: 24px;
  font-size: 13px;
  line-height: 20px;
}

.c0[data-has-icon] span {
  position: absolute;
  top: -8px;
  left: 10px;
}

.c0[data-has-icon] svg {
  position: absolute;
  z-index: -1;
  top: 0;
}

<div
  class="c0"
  data-has-icon="true"
  data-notification-type="small"
>
   
  <div>
    <img
      src="https://source.unsplash.com/random/16x16"
    />
  </div>
</div>
`;
